common4j/src/main/com/microsoft/identity/common/java/challengehandlers/IDeviceCertificate.java

```diff
@@ -47,27 +47,22 @@ public interface IDeviceCertificate {
      *
      * @return {@link X509Certificate}
      */
-    X509Certificate getCertificate();
-
-    /**
-     * Gets a private key.
-     *
-     * @return private key
-     */
-    PrivateKey getPrivateKey();
+    @NonNull
+    X509Certificate getX509();
 
     /**
-     * Gets a public key.
+     * Gets thumbPrint for certificate.
      *
-     * @return RSA public key.
+     * @return thumbPrint for certificate.
      */
-    PublicKey getPublicKey();
+    String getThumbPrint();
 
     /**
-     * Gets thumbPrint for certificate.
+     * Signs a piece of data with the (private key associated to the) certificate.
      *
-     * @return thumbPrint for certificate.
+     * @param algorithm         algorithm for signing the data.
+     * @param dataToBeSigned    the data to be signed.
      */
-    String getThumbPrint();
+    byte[] sign(@NonNull final String algorithm, final byte[] dataToBeSigned) throws ClientException;
```

Comment:
nit : dataToBeSigned should also be non-null?

Reply:
it should not be, but we cannot add NonNull annotation to a byte array (it will be ignored by lombok).